src/docs/sdk/unified-api/tracing.mdx

```diff
@@ -0,0 +1,60 @@
+---
+title: 'Guideline for AM support'
+---
+
+This document covers how unified SDKs should implement AM.
+Things listed here are in no specific order, cover the API shape and not how it should be internally implemented.
+Read this fully before asking questions.
+
+See [https://github.com/getsentry/sentry-javascript](https://github.com/getsentry/sentry-javascript) 
+
+or
+
+[https://github.com/getsentry/sentry-javascript/tree/master/packages/apm](https://github.com/getsentry/sentry-javascript/tree/master/packages/apm) for a reference implementation
+
+- New option for the SDK `tracesSampleRate`
+    - Defaults to `0.0`, which means send no transactions (0% delivery rate)
+    - `1.0` (100% delivery rate) means send all transactions
+    - Value is expressed in `0.0` - `1.0` float, ranged from 0% to 100% rate
+    - Default makes AM an opt-in feature
+- Introduce `Sentry.startTransaction`
+    - Internally this creates a `Transaction` and returns the instance
+    - Users need to interaction with the instance and keep track of it themselves
+    - Internally we can use the `Scope` to hold on the a reference of it if needed
+- Introduce the new type `Span` → [https://develop.sentry.dev/sdk/event-payloads/span/](https://develop.sentry.dev/sdk/event-payloads/span/)
+- Introduce new type `Transaction` which inherits from `Span`
+- `Span` Interface
+    - When a `Span` is created, set the `startSimestamp` to the current time
+    - `SpanContext` is the attribute collection for a `Span` (Can be an implementation detail)
+    - The relation between parent - child is captured in the property `parentSpanId`
+- `Transaction` Interface
+    - A `Transaction` internally holds a flat list of child Spans (not a tree structure)
+    - `Transaction` has additionally a `setName` method the set the name of the transaction
+    - `Transaction` receives a `TransactionContext` on creation (new property vs. `SpanContext` is `name`)
+    - Since a `Transaction` inherits a `Span` it has all functions available and can be interacted with like it was a `Span`
+- `Hub` → Introduce a method called `startSpan`
+    - Creates a new `Span` instance
+    - If there is already a `Span` on the current `Scope`, the created `Span` should be its child
+    - This method deals with sampling, and therefore it should take the `tracesSampleRate` option into account:
+        - If it's a child `Span`, we inherit the decision from the parent
+        - If it's a root span, we need to roll the dice here
+        - Depending on the outcome, the sample decision should be stored in the `Span`'s `sampled` property and again forwarded to its children
+- `Hub` → Introduce a method called `traceHeaders`
+    - This function returns a header (string) `sentry-trace`
+    - The value should be the trace header string of the `Span` that is currently on the `Scope`
```

Comment:
This is only relevant for inter-integration communication. In JS we don't use it anywhere (right now) but for python we need it to attach this info for background queue integration.

And that doesn't mean we need to add it since the Spans also have the `toTraceparent` function which should be used to attach to outgoing requests.